packages/arb-token-bridge-ui/src/components/Sidebar/SidebarMenu.tsx

```diff
@@ -0,0 +1,387 @@
+
+        <ArrowTopRightOnSquareIcon className="mr-3 h-4 w-4 opacity-50 group-hover:opacity-100" />
+      </ExternalLink>
+    )
+  }
+
+  if (!isExternalLink && link) {
+    return (
+      <Link href={link} className={subMenuClasses} onClick={onClick}>
+        {children}
+      </Link>
+    )
+  }
+
+  return (
+    <button className={subMenuClasses} onClick={onClick}>
+      {children}
+    </button>
+  )
+}
+
+export const SidebarMenu = ({
+  menuItemClickCallback,
+  className
+}: {
+  menuItemClickCallback?: () => void
+  className?: string
+}) => {
+  const posthog = usePostHog()
+  const { sidebarOpened, activeMenu, setActiveMenu } = useSidebarStore()
+
+  function sendClickEventForGroup(groupTitle: string) {
+    posthog?.capture('Sidebar Menu Dropdown Click', {
+      Group: groupTitle
+    })
+  }
+
+  function sendClickEventForLink(linkTitle: string) {
+    posthog?.capture('Sidebar Menu Link Click', {
+      Link: linkTitle
+    })
+  }
+
+  function sendClickEventForMenuItem(menuItem: string) {
+    posthog?.capture('Sidebar Menu Item Click', {
+      Item: menuItem
+    })
+  }
+
+  const toggleActiveMenu = (menuId: string) => {
+    if (activeMenu === menuId) {
+      setActiveMenu('')
+    } else {
+      setActiveMenu(menuId)
+    }
+
+    // also send analytics event for group toggle click
+    const menuGroupName = menuConfig.find(menuItem => menuId === menuItem.id)?.[
+      'title'
+    ]
+    if (menuGroupName) sendClickEventForGroup(menuGroupName)
+  }
+
+  const menuConfig: MenuConfig[] = [
+    {
+      id: 'home',
+      title: 'Home',
+      iconSrc: IconHome,
+      isExternalLink: false,
+      isExpandable: false,
+      link: PORTAL_DOMAIN,
+      onClick: () => {
+        setActiveMenu('home')
+        menuItemClickCallback?.()
+      }
+    },
+    {
+      id: 'projects',
+      title: 'Projects',
+      iconSrc: IconProjects,
+      isExternalLink: false,
+      isExpandable: false,
+      link: PORTAL_DOMAIN,
+      onClick: () => {
+        toggleActiveMenu('projects')
+      }
+    },
+    {
+      id: 'missions',
+      title: 'Missions',
+      iconSrc: IconMissions,
+      isExternalLink: false,
+      link: `${PORTAL_DOMAIN}/missions`,
+      onClick: () => {
+        menuItemClickCallback?.()
+        sendClickEventForLink('Missions')
+      }
+    },
+    {
+      id: 'bridge',
+      title: 'Bridge',
+      iconSrc: IconBridge,
+      isExternalLink: false,
+      onClick: () => {
+        menuItemClickCallback?.()
+        sendClickEventForLink('Bridge')
+        toggleActiveMenu('bridge')
+      }
+    },
+    {
+      id: 'learn',
+      title: 'Learn',
+      iconSrc: IconLearn,
+      isExpandable: true,
+      isExternalLink: false,
+      onClick: () => toggleActiveMenu('learn'),
+      children: (
+        <div className="mb-4 flex flex-col gap-2 text-sm">
+          {learnMenuItems.map(subMenuItem => (
+            <SubMenuItem
+              key={stringToKey(subMenuItem.title)}
```

Comment:
Do we need `stringToKey` at all? It's not any different to using `subMenuItem.title` directly.

Reply:
that's from portal
i am not touching it hehe